### PR TITLE
cgmath version 0.18 and automatic arcball camera placement

### PR DESCRIPTION
**Pull request by beltegeuse**
Hi Will, 

Please find a pull request that implements two changes:
1) Bump cgmath to last version (0.18). 
2) Compute AABB from the loaded obj to automatically place the camera for obj_viewer and obj_ao_parallel. This changes the support `display` struct and adds a new `aabb` struct.

Best

### CODE REVIEW

File: examples/support/Cargo.toml [hunk 1]
```diff
@@ -5,8 +5,8 @@ authors = ["Will Usher <willusher.life@gmail.com>"]
 
 [dependencies]
 image = "0.22.0"
-arcball = "1.0.0"
-cgmath = "0.17.0"
+arcball = {path = "../../../arcball" }
```
Comment (Twinklebear) [inline examples/support/Cargo.toml:8]: I think the CI failures are from here where it's looking for arcball on a local path. I think we could increment it to use arcball 1.1.0, but are there some other changes you had locally that the rest of the PR changes relied on?

**Review by Twinklebear — CHANGES_REQUESTED** [review]
Thanks @beltegeuse , this looks good! Having automatic placement of the camera will be great for the demo apps. I think the only change needed is how the arcball crate is being included.

I've also been (slowly) working on refactoring the codebase a bit to fix the lifetime annoyances, and to also move the viewer to a wgpu-rs based one, since glium is in maintenance mode and I was having some issues with it and related libraries.

File: examples/support/Cargo.toml (hunk 1 above)
Comment (beltegeuse) [reply examples/support/Cargo.toml:8]: Indeed, good catch. I did not remember that I have done this modification :) Sorry about that!

**Review by Twinklebear — APPROVED** [review]
Thanks @beltegeuse , this looks great!

**Pull request merged**